### PR TITLE
feat(chungthuang): Periodic fetch Google Meet reports

### PR DESCRIPTION
**Pull request by chungthuang**

### CODE REVIEW

File: api/google-manager/services/google-manager.js [hunk 1]
```diff
@@ -103,6 +105,27 @@ class GoogleManager {
     }
   }
 
+  async listAuditReports(userKey, applicationName, eventName) {
+    const auth = this.auth;
+    const service = google.admin({ version: 'reports_v1', auth });
```
[the remark this answers is not in the document]
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:110]: It's a function to call the admin SDK https://developers.google.com/admin-sdk. `reports_v1` means we want to use the Reports API under admin SDK https://developers.google.com/admin-sdk/reports/reference/rest.

File: api/google-manager/services/google-manager.js [hunk 2]
```diff
@@ -103,6 +105,27 @@ class GoogleManager {
...
+    const currentTime = new Date();
+    const startTime = new Date(currentTime - this.auditFreqMilliSecs).toISOString();
+    try {
+      const resp = await service.activities.list({
```
[the remark this answers is not in the document]
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:114]: `activities.list` calls this API https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list. `resp` is a variable that stores the response. The expected format of the response is https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list.

File: api/google-manager/services/google-manager.js [hunk 3]
```diff
@@ -128,8 +155,20 @@ if (!isDevEnv()) {
   const privateKey = googleKey.private_key;
   const groupID = process.env.DEVLAUNCHERS_GOOGLE_DIRECTORY_GROUP_ID;
   const serverBaseURL = process.env.URL;
+  const auditFreq = process.env.AUDIT_FREQ_MINUTES;
+
+  const manager = new GoogleManager(email, privateKey, groupID, serverBaseURL, auditFreq);
+  const cronTime = `0 */${auditFreq} * * * *`;
+  const job = new CronJob(cronTime, function () {
+    manager.listAuditReports('all', 'meet', 'call_ended').then(reports => {
```
[the remark this answers is not in the document]
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:163]: They are parameters to `listAuditReports(userKey, applicationName, eventName)`. The function calls https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list. User key is the path parameter defined in https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list. `applicationName` is one of https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list#applicationname. And `eventName` is a query parameter https://developers.google.com/admin-sdk/reports/reference/rest/v1/activities/list#query-parameters.

[the remark this answers is not in the document]
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:163]: `listAuditReports('all', 'meet', 'call_ended')` means we want to get all `call_ended` event for the Google Meet application for all users.

File: api/google-manager/services/google-manager.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+const { CronJob } = require('cron');
 const fs = require('fs');
 const { JWT } = require('google-auth-library');
```
Comment (jiawen195) [inline api/google-manager/services/google-manager.js:3]: Is JWT a library, if it is, what does new JWT() mean?

File: api/google-manager/services/google-manager.js [hunk 5]
```diff
@@ -128,8 +155,20 @@ if (!isDevEnv()) {
...
+    manager.listAuditReports('all', 'meet', 'call_ended').then(reports => {
+      reports.items.forEach(r => {
+        strapi.log.info('Report for Google Meet', r.events[0].parameters);
+      });
+    });
+  });
+  job.start();
 
-  module.exports = new GoogleManager(email, privateKey, groupID, serverBaseURL);
+  module.exports = manager;
```
Comment (jiawen195) [inline api/google-manager/services/google-manager.js:171]: What is exports?

File: api/google-manager/services/google-manager.js (hunk 4 above)
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:3]: It is a class in the google-auth-library https://cloud.google.com/nodejs/docs/reference/google-auth-library/latest/google-auth-library/jwt.

File: api/google-manager/services/google-manager.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:114]: We only pass in one applicatioName, so the response is limited to one. The parameters we passed in are `manager.listAuditReports('all', 'meet', 'call_ended')`.

[the remark this answers is not in the document]
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:114]: `startTime = new Date(currentTime - this.auditFreqMilliSecs).toISOString()`. `this.auditFreqMilliSecs` is configured to an hour, so this means we want the report in the last hour.

File: api/google-manager/services/google-manager.js (hunk 5 above)
Comment (chungthuang) [reply api/google-manager/services/google-manager.js:171]: So this service can be used in different modules.

**Pull request merged**